### PR TITLE
metasploit: 6.3.43 -> 6.3.44

### PR DESCRIPTION
**Pull request by fabaff**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 269842` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>metasploit</li>
    <li>msfpc</li>
    <li>social-engineer-toolkit</li>
  </ul>
</details>

**Review by bjornfor — COMMENTED** [review]
Please split refactoring and version bumps into separate commits.

File: pkgs/tools/security/metasploit/default.nix
Comment (bjornfor) [inline pkgs/tools/security/metasploit/default.nix:33]: 9 lines of unrelated refactoring.

Comment (makefu) [reply pkgs/tools/security/metasploit/default.nix:33]: Is this an issue? I often get encouraged to do this kind of refactoring as part of a package update ( e.g. https://github.com/NixOS/nixpkgs/pull/266533#discussion_r1398411940 ).

Comment (bjornfor) [reply pkgs/tools/security/metasploit/default.nix:33]: When I read a commit that bumps a version, I expect it to only do that. My mind is focused on "what's the required changes needed to do the update?". If there's unrelated changes, that's distracting. Also, version bumps pull in new code from the internet, which I think requires more careful review than reformatting changes.

Reformatting in a version bump PR is fine of course, but I think they should be a separate commit, where one can say what is done and why. (In this case, I don't even know why those 9 lines are changed. Is it an automated tool that did it? Or did the author do it manually because they think it improves readability?)

Does it make sense? Thoughts?

> I often get encouraged to do this kind of refactoring as part of a package update ( e.g. https://github.com/NixOS/nixpkgs/pull/266533#discussion_r1398411940 ).

Isn't that just about "sha256" -> "hash"? The line gets changed anyway in a version bump, so I think that's OK.

Comment (fabaff) [reply pkgs/tools/security/metasploit/default.nix:33]: > Isn't that just about "sha256" -> "hash"? The line gets changed anyway in a version bump, so I think that's OK.

Technically are changing `sha256` to `hash` and the SRI hash itself two changes. But that's usually combined like adding `refs/tags/` to avoid branch/tag issue when updating. The update bot is doing this anyways.

I think that it's a good trade-off to include formatting changes, high level changes and alike in updates as this will keep the amount of commits low and the maintenance work-flow straight forward and time-efficient. Of course, one could always squash the commit before merging (as we don't let GitHub do this) but would add additional tasks on the reviewer's plate.

IMHO, in regard of back-porting the argument of keeping the commits atomic weighted much more that changes which go to `master` only.

For sure, the line is a bit blurry what changes or amount of changes should go in one commit. Thus, it's good that a reviewer can always ask to split as the committer's opinion and the reviewers can differ.

> Is it an automated tool that did it?

Was done manually as `nixpkgs-fmt` will not go that far.

> Or did the author do it manually because they think it improves readability?

Readability? Perhaps, as always, that depends on the point of view. It was done to follow our code conventions a little closer and also to have easier to read diffs in the future. Short lists or attribute sets can be written on one line but this can't be alphabetically sorted by default tools.

**Review by makefu — APPROVED** [review]
tested msfconsole, lgtm

**Pull request merged**
